### DISCUSSION TITLE
Defect: Dependency scope propagation is broken

### DISCUSSION THREAD

**Discussion by kristjanvalur** [fastapi/fastapi#8408] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# conceptual excerpt from unittests of associated PR
async def security1(scopes: SecurityScopes):
    return _security1(scopes.scopes)

async def security2(scopes: SecurityScopes):
    return _security2(scopes.scopes)

async def dep3(
    dep=Security(security1, scopes=["scope1"]),
    dep2=Security(security2, scopes=["scope2"]),
):
    return {}


@pytest.fixture
def mocks():
    with patch.dict(globals(), {"_security1": Mock()}):
        with patch.dict(globals(), {"_security2": Mock()}):
            yield


app = FastAPI()

@app.get("/recursive_scopes")
def recursive_scopes(dep=Security(dep3, scopes=["scope3"])):
    return {}

# we expect `security1` to be called with scopes `["scope3, "scope1"]` and `security2` with `["scope3", "scope2"]`
```

### Description

The propagation of `Security` `scopes` is broken.  Scopes of dependencies are inherited by
sub-dependencies, adding their own scopes.  A case of parameter mutation in FastAPI causes leakage
of added scopes to sibling dependencies.  Please see associated PR #5624 for information.

### Operating System

Linux

### FastAPI Version

main

### Python Version

3.10

**Comment by kristjanvalur** [comment]
I see a number of issues, PRs, being converted to discussions, without any comments, what is going on?

**Comment by tiangolo** [comment, marked as answer]
This was solved by https://github.com/fastapi/fastapi/pull/5624, thank you! :raised_hands:
